framework/include/base/Assembly.h

```diff
@@ -458,7 +458,11 @@ class Assembly
    * Returns the reference to the transformed jacobian weights on a current face
    * @return A _reference_.  Make sure to store this as a reference!
    */
-  const MooseArray<Real> & JxWNeighbor() const { return _current_JxW_neighbor; }
+  const MooseArray<Real> & JxWNeighbor() const
+  {
+    _need_JxW_neighbor = true;
+    return _current_JxW_neighbor;
+  }
```

Comment:
Move to source

Reply:
This is consistent with the style of other functions in `Assembly.h` and it should not be an issue.

Reply:
Just because we do it on other places doesn't mean it's good practice. Please move it. I just cleaned up ScalarVariable with the same.

Reply:
I also do not want to move it because of inlining. I do not know what will happen with compilers and do not want to guess. Keeping short functions in header file is the way to go imo.

Reply:
I personally am ok with this, but (and I know we haven't been good about this historically) at a minimum I want it moved out of the class definition for people wanting to quickly survey the `Assembly` APIs. Folks shouldn't have to skip over code blocks in the class definition

Reply:
Looks like this requested change hasn't been addressed yet

Reply:
If we want this change, I suggest we do this together with other functions. I do not make this change not because I am lazy. It is because I have strong opinions on this. If someone want to override my decision in the future, it is fine but that is not my decision.

Reply:
We have spent more time discussing this than it would take to make the change. Regardless of old allowance of code quality, we should be taking steps to increase quality and this is one of them. Changed code should be held to these standards. Please move it.

Reply:
I have spent too many word to make my position clear...

Reply:
I believe our position is clear.